src/mpi/romio/adio/ad_daos/ad_daos_io_str.c

```diff
@@ -154,15 +154,16 @@ ADIOI_DAOS_StridedListIO(ADIO_File fd, const void *buf, MPI_Aint count,
 
     /* Create Memory SGL */
     file_length = 0;
+    MPI_Count k; // how many entries in scatter-gather list
     if (!buftype_is_contig) {
         flat_buf = ADIOI_Flatten_and_find(datatype);
         mem_list_count = count * flat_buf->count;
 
         iovs = (d_iov_t *) ADIOI_Malloc(mem_list_count * sizeof(d_iov_t));
 
         k = 0;
-        for (j = 0; j < count; j++) {
-            for (i = 0; i < flat_buf->count; i++) {
+        for (MPI_Count j = 0; j < count; j++) {
+            for (MPI_Countl = 0; i < flat_buf->count; i++) {
```

Comment:
Typo?

Reply:
indeed.  In fact I will probably need to patch the whole daos driver once our daos machine comes back.   Well... let me see if I can at least compile test on our daos machine.

Reply:
I have at least compile-tested daos. not only does it compile but I found a few types I needed to promote.